src/iop/atrous.c

```diff
@@ -187,21 +178,35 @@ static inline void weight(const float *c1, const float *c2, const float sharpen,
  * wc = exp(-sharpen*(SQR(c1[1] - c2[1]) + SQR(c1[2] - c2[2]))
  *    = exp(-s*(d2+d3)) (as noted in code comments below)
  */
-static inline __m128 weight_sse2(const __m128 *c1, const __m128 *c2, const float sharpen)
+static inline __m128 weight_sse2(const __m128 c1, const __m128 c2, const float sharpen)
 {
-  const __m128 vsharpen = _mm_set1_ps(-sharpen); // (-s, -s, -s, -s)
-  __m128 diff = _mm_sub_ps(*c1, *c2);
-  __m128 square = _mm_mul_ps(diff, diff);                                   // (?, d3, d2, d1)
-  __m128 square2 = _mm_shuffle_ps(square, square, _MM_SHUFFLE(3, 1, 2, 0)); // (?, d2, d3, d1)
-  __m128 added = _mm_add_ps(square, square2);                               // (?, d2+d3, d2+d3, 2*d1)
-  added = _mm_sub_ss(added, square);                                        // (?, d2+d3, d2+d3, d1)
-  __m128 sharpened = _mm_mul_ps(added, vsharpen);                   // (?, -s*(d2+d3), -s*(d2+d3), -s*d1)
-  __m128 exp = dt_fast_expf_sse2(sharpened);                        // (?, wc, wc, wl)
+  const __m128 vsharpen = _mm_set1_ps(-sharpen);                    // (-s, -s, -s, -s)
+  const __m128 diff = (c1 - c2);
+  const __m128 square = diff * diff;                                // (?, d3, d2, d1)
+  const __m128 square2 = _mm_shuffle_ps(square, square, _MM_SHUFFLE(3, 1, 2, 0)); // (?, d2, d3, d1)
+  const __m128 added = square + square2 - square;                   // (?, d2+d3, d2+d3, d1)
```

Comment:
@aurelienpierre May I ask what this is for?
Is this a trick for clipping small values of square2 to zero?

So for example 100000000 + 0.000001 - 100000000 = 0

Reply:
@heckflosse I'm not the author of that code, so I think @hanatos will answer better than me. I just made the loops a tad faster here.

Reply:
@aurelienpierre Ok, I just was curious because at least on first sight calculating `a = b + c - b` does not make much sense

Reply:
damit ! you are right ! so easy to confuse  `_mm_sub_ps` and `_mm_sub_ss`… Thanks !